### PR TITLE
Spelling: "U2F", moved …

### PR DESCRIPTION
**Pull request by comradekingu**

### CODE REVIEW

File: js/settingsview.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/* global Backbone, Handlebars, OC, u2f, Promise, _ */
+/* global Backbone, Handlebars, OC, U2F, Promise, _ */
```
Comment (ChristophWurst) [inline js/settingsview.js]: revert this change please. This comment is about the global `u2f` variable.

File: js/settingsview.js [hunk 2]
```diff
@@ -36,7 +36,7 @@
 		+ '     <span><small>' + t('twofactor_u2f', 'You can add as many devices as you like. It is recommended to give each device a distinct name.') + '</small></span>'
 		+ '	{{else}}'
 		+ '     <span class="icon-loading-small u2f-loading"></span>'
-		+ '	<span>' + t('twofactor_u2f', 'Adding a new device …') + '</span>'
+		+ '	<span>' + t('twofactor_u2f', 'Adding a new device…') + '</span>'
```
Comment (ChristophWurst) [inline js/settingsview.js]: According to @jancborchardt there should be a space before '…'

Comment (comradekingu) [inline js/settingsview.js]: Whatever the paradigm, it must be consistent across the whole project. Right now it isn't.
Not that it is a pressing issue, given there are more things to handle, but I think 
*… makes for fewer errors in translation.

Sticking to one way, and not invoking or otherwise require thought on part of the translators would be ideal.

To elaborate on the topic, this is one of the strings:
> Smile in 3… 2… 1!

One could about that, "what is that…"  or by writing  
"Then there is this "Smile in 3… 2… 1!" …" To maybe signify the confusion there being a thing onto itself.
I think the strict use there would be to say * […]

However, luckily there isn't any of that going on in Nextcloud.

In Nextcloud, ellipsis is invariably used to note 
an action/verb in infinitive tense "Hide description…"
preteritum-  "Verifying…"
or 
as an omission "Server address https://…"

A common use, described in some style-guides, and employed in some languages, is the use of ellipsis to replace words. This becomes an issue with things like "Announcing" (which is an actual string) followed by *… means the action, or in the case of * … "Announcing …" would signify a subjective being announced.

The problem then becomes:
"This i can not see…" Where it could be interpreted as unseen, or possibly not see-sawable.
(A see-saw being a type of saw.) Which isn't a problem if you aren't extremely unlucky and unaware.
This is possibly why in the strictest sense, you would do …. ellipsis with trailing dot, to mark an end an end of a sentence where the intention was to first append, and then end.

As long as you avoid words that could feasibly be added onto by the use of an ellipsis, *… aligns better with interpretation and the use in Nextcloud.

Comment (jancborchardt) [inline js/settingsview.js]: It's quite simple. If there is an incomplete word like https://, the ellipsis comes directly after. If there is no incomplete word, there should be a space. Hence please add the space here again. :)

**Comment by comradekingu** [comment]
@jancborchardt 
That is unfortunately not the only way it is used. If your idea of not replacing things is "word …" then you are at odds with people who don't. And systems that don't.

Lets see what makes more sense, seeing as both are valid in English.

If they only use "word …" for replacing whole words in a sentence, like "Canada … a country" (Canada is a country) and otherwise "word…" "Canada is a country …" (Canada is a country in the northern hemisphere) now actions look like they are only spelled out in part. That is the lowest common denominator.

If you wanted to sound moronic, or question the legitimacy of Canada, you would do:
"Canada is a country…" But what if you _only_ do that to replace parts of a word?

Lets say those people didn't exist, apply for sense, or complicate matters if need be:

The use of ellipsis for irony/pause for thought/effect, is the more common use.

So you are looking at misinterpreting in a word-ending, or a sentence progression.
Respectively with "word…" or "word …" for _not_ replacing words.

Replacing parts of words is the less common, of the less common use.
For which almost nobody is going to do "[…]" "…." or ".../…" consistently for replacement/interpretation or effect. And fewer still will get it right.
Even less often, they will be able to go about their business alone, so they run into simpletons doing any of the others. But yes, if everyone did even more markup, everything would be fine. But it would be a stupid system for reasons you could replace with something like _____ instead.

So in this case, there is nothing wrong with KISS. There _is_ actually something wrong with the other systems than "word…" in other languages.

OK, so what is the problem:

You would only do "word…" it to bleep out parts of words, which you could do anyway. It would be useful if you wanted to signify two valid and obvious interpretations, but you could do that anyhow with "word…" word-play and word-feude for example.

"Over the use of … much word…"  :)

What I am getting at is you can always add onto a sentence with something, that is where more errors are interpreted in.

Short of French, punctuation is always "word*" Which is the real issue here, "word…" is allowed, whereas "word …" sometimes isn't. it makes for fewer errors in translation.

"word …" is not consistent with action-use in software. Strike two.

Nobody cared enough to do anything about tripledot being used in place of ellipsis, but the positioning is all of a sudden very important. Just an observation, but OK.
So the argument is, for reasons why "consistency" sounds good, it is actually a quality the Nextcloud codebase features.

Current tally: (Ordered by levels of trouble).

35…   Ellipsis without space
32 …  Ellipsis preceded by space
42...   Tripledot without
22 ...  Tripledot preceded by space
  4....  Quadrupledot

**Comment by jancborchardt** [comment]
> Nobody cared enough to do anything about tripledot being used in place of ellipsis, but the positioning is all of a sudden very important. Just an observation, but OK.

I can’t be everywhere, and to many people an ellipsis just looks like a tripledot, or they don’t care about the space. The point is that space followed by ellipsis is the stylistic choice we made. Everything else is just how people individually decided to do it.

So if you want to fix it, please do it the way we want it to have. which is:
- Space and ellipsis when it’s after words: `Loading …`
- Just an ellipsis when it’s after an incomplete word: `https://…`

Thank you.

**Comment by ChristophWurst** [comment]
Closing this PR as it has been inactive for a few months. Thanks.

**Comment by jancborchardt** [comment]
Just for the record: the case change to U2F us ok, so best to keep that separate. :)
